src/main/java/de/retest/web/screenshot/FullPageScreenshot.java

```diff
@@ -11,25 +11,31 @@
 import com.assertthat.selenium_shutterbug.core.Shutterbug;
 import com.assertthat.selenium_shutterbug.utils.web.ScrollStrategy;
 
+import lombok.extern.slf4j.Slf4j;
+
 /**
  * Scrolls to the bottom of the page and takes a screenshot of the entire page.
  */
+@Slf4j
 public class FullPageScreenshot implements ScreenshotProvider {
 
 	private static final int SCROLL_TIMEOUT_MS = 100;
 	private static final boolean USE_DEVICE_PIXEL_RATIO = true;
 
 	@Override
 	public BufferedImage shoot( final WebDriver driver ) {
+		final long startTime = System.currentTimeMillis();
 		if ( driver instanceof ChromeDriver ) {
 			final BufferedImage image = Shutterbug //
 					.shootPage( driver, ScrollStrategy.WHOLE_PAGE_CHROME, SCROLL_TIMEOUT_MS, USE_DEVICE_PIXEL_RATIO ) //
 					.getImage();
 			return resizeImage( image, image.getWidth() / SCALE, image.getHeight() / SCALE );
 		}
-		return Shutterbug //
+		final BufferedImage result = Shutterbug //
 				.shootPage( driver, ScrollStrategy.BOTH_DIRECTIONS, SCROLL_TIMEOUT_MS, USE_DEVICE_PIXEL_RATIO ) //
 				.getImage();
+		log.info( "Took {}ms to create the screenshot.", System.currentTimeMillis() - startTime );
```

Comment:
You do not profile in every case (i.e. `ChromeDriver`). If you really want to do profiling of the screenshots (and I currently do not see why you would), I suggest to do that in `ScreenshotProviders#shoot` in a central place and keep these implementations clean.

Reply:
Done